### PR TITLE
Address warnings, including attention mask issue that changes the recs

### PR DESCRIPTION
**Pull request by karlhigley**
This addresses three warnings of varying importance:
* The LK pipeline code warned that `SoftmaxSampler` didn't have an annotated return type
* The HF Tokenizers library warned about a tokenization space clean-up behavior that has been enabled by default but will be disabled in future versions
* The HF Transformers library warned about the absence of an attention mask being passed to DistilBERT, which can (and does!) affect the article embeddings

### CODE REVIEW

**Comment by karlhigley** [comment]
@sophiasun0515 The change that adds an attention mask to the DistilBERT input changes eventual NRMS output ranking, which makes me wonder if we should retrain the model with the attention mask in place. Thoughts?

**Comment by mdekstrand** [comment]
@karlhigley the test failure should be resolved now with a rerun (I fixed the shared data publication on `main`).

**Review by mdekstrand — APPROVED** [review]

**Review by sophiasun0515 — APPROVED** [review]
Looks good. Thanks for fixing this!
